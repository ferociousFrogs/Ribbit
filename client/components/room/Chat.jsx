import React from 'react';

class Chat extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="row border right-side">
        <div className="chat-header">
          <span className="glyphicon glyphicon-user" />
          <span>Chat with USER</span>
        </div>
        <div className="chat-area">
          chat area
        </div>
        <div className="chat-input">
          <form>
            <input
              type="text"
              className="col-md-10 col-sm-10 col-xs-10"
              placeholder="Type your message here."
            />
            <button className="btn-info col-md-2  col-sm-2 col-xs-2">Send</button>
          </form>
        </div>
      </div>
    );
  }
}

export default Chat;
