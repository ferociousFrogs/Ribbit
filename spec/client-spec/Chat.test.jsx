import { renderIntoDocument } from 'react-dom/test-utils';
import React from 'react';
import renderer from 'react-test-renderer';
import Chat from '../../client/components/room/Chat';

test('Chat component snapshot test', () => {
  const component = renderer.create(<Chat />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Chat component should be a stateful class component', () => {
  expect(React.Component.isPrototypeOf(Chat)).toEqual(true);
});
