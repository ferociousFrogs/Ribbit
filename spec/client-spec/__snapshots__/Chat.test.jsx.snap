// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component snapshot test 1`] = `
<div
  className="row border right-side"
>
  <div
    className="chat-header"
  >
    <span
      className="glyphicon glyphicon-user"
    />
    <span>
      Chat with USER
    </span>
  </div>
  <div
    className="chat-area"
  >
    chat area
  </div>
  <div
    className="chat-input"
  >
    <form>
      <input
        className="col-md-10 col-sm-10 col-xs-10"
        placeholder="Type your message here."
        type="text"
      />
      <button
        className="btn-info col-md-2  col-sm-2 col-xs-2"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;
